addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -412,17 +412,18 @@ private void processApplicationStateChanged(String method, JsonObject json) {
 
             this.volume = volume;
         } else {
-            logger.debug("Unknown event from kodi {}: {}", method, json.toString());
+            logger.debug("Unknown event from Kodi {}: {}", method, json.toString());
         }
+        listener.updateConnectionState(true);
     }
 
     private void processSystemStateChanged(String method, JsonObject json) {
-        if ("System.OnQuit".equals(method) || "System.OnSleep".equals(method) || "System.OnRestart".equals(method)) {
-            try {
-                listener.updateConnectionState(false);
-            } catch (Exception e) {
-                logger.error("Event listener invoking error", e);
-            }
+        if ("System.OnQuit".equals(method) || "System.OnRestart".equals(method) || "System.OnSleep".equals(method)) {
+            listener.updateConnectionState(false);
```

Comment:
@martinvw I am glad you are asking. I thought about it and analysed the situation. Let me explain my doing:
1. There are a lot more lines of code which calls the same `updateConnectionState` method which aren't wrapped by a try catch block.
2. The definition of the method doesn't have a `throw`. So why catch something, which doesn't throw anything.
3. If you have a look at the implementation of the method you will see, that in the `false` part there is only a `updateStatus(ThingStatus.OFFLINE)`. This method needs no try catch block. And in the `true` part there is a `updateStatus(ThingStatus.ONLINE)` followed by another try catch block. Double usage.